[14.0] [IMP] ckeditor: Make todoList available

lieblinger opened the pull request:
This PR adds a converter for TodoList. 

The default schema for TodoLists from CKEditor is not supported by Odoo, cause <input> elements get stripped away.

My PR adds functions to convert the TODOList schema back and forwards (Odoo <-> CKEditor). By default this PR saves Odoo's schema to the database and converts it back to the CKEditor schema for the frontend. This allows to keep TODOList working, even if someone switches back to Odoo's default editor.

What is not working: With Odoo's editor it was possible to check boxes in ReadOnlyView. This is not implemented by me and also not really needed in my case.

lieblinger committed:
[14.0] [IMP] ckeditor: Make todoList available

ivantodorovich commented:
Hello, thanks a lot for this!

I tested it and it works 🎉 

I would rather keep the same Odoo core style, if possible. The scss you added don't really look like the "standard" to-do list, do you think that's possible?:

![image](https://user-images.githubusercontent.com/1914185/199075891-79d1a883-a894-4ce9-ba25-83f9ab188766.png)


Also, please check pre-commit, as it's failing. ([for more information](https://odoo-community.org/blog/news-updates-1/black-isort-pre-commit-97))

tarteo reviewed (approved):
Awesome thanks!
![image](https://user-images.githubusercontent.com/10028499/202202605-858e65f9-5414-4a96-85e5-4d9e6d4b41a4.png)

lieblinger committed:
[14.0][IMP] web_widget_ckeditor: Improve code to fix pre-commit

[14.0][IMP] web_widget_ckeditor: Improve code to fix pre-commit - 2

[14.0][IMP] web_widget_ckeditor: Improve code to fix pre-commit - 3

ivantodorovich commented:
Hello @lieblinger 

I see you're struggling with pre-commit.

I recommend you to install it locally, as it will fix most of the "issues" automatically for you, and at the very least save you lots of time by running the checks in your terminal.

You can install it following the official docs: https://pre-commit.com/#install
Then, with a shell on your git clone, run `pre-commit install`. That will install the pre-commit hooks into your local clone.
The hooks will run before every new commit, but since you have already unstyled commits, you might need to run `pre-commit run -a` to scan the whole repo

dreispt commented:
/ocabot merge patch

bosd commented:
@lieblinger Can you squash the last 3 commits?

dreispt commented:
@lieblinger Plase squash your commits
